semgrep-core/matching/Matching_generic.mli

```diff
@@ -62,6 +62,7 @@ val check_and_add_metavar_binding :
 (*s: signature [[Matching_generic.has_ellipsis_stmts]] *)
 val has_ellipsis_stmts : AST_generic.stmt list -> bool
 (*e: signature [[Matching_generic.has_ellipsis_stmts]] *)
+val inits_and_rest_of_list: 'a list -> ('a list * 'a list) list
```

Comment:
The more I think about this, the more I think we need to include the delimiter in the match